src/main/java/com/minecolonies/coremod/placementhandlers/HutPlacementHandler.java

```diff
@@ -64,31 +65,35 @@ public ActionProcessingResult handle(
                 final BlockEntity be = world.getBlockEntity(pos);
                 if (be != null)
                 {
+                    final StructurePackMeta pack = StructurePacks.getStructurePack(blueprint.getPackName());
                     if (pos.equals(centerPos))
                     {
-                        final String location = StructurePacks.getStructurePack(blueprint.getPackName()).getSubPath(blueprint.getFilePath().resolve(blueprint.getFileName()));
+                        final String location = pack.getSubPath(blueprint.getFilePath().resolve(blueprint.getFileName()));
                         ((IBlueprintDataProviderBE) be).setBlueprintPath(location);
                     }
-                    else if(!((IBlueprintDataProviderBE) be).getPositionedTags().getOrDefault(BlockPos.ZERO, Collections.emptyList()).contains("invisible"))
+                    else if (!((IBlueprintDataProviderBE) be).getPositionedTags().getOrDefault(BlockPos.ZERO, Collections.emptyList()).contains("invisible"))
                     {
                         final String partialPath;
                         if (((IBlueprintDataProviderBE) be).getSchematicName().isEmpty())
                         {
                             final String[] elements = ((IBlueprintDataProviderBE) be).getBlueprintPath().split("/");
-                            partialPath = StructurePacks.getStructurePack(blueprint.getPackName()).getSubPath(blueprint.getFilePath().resolve(elements[elements.length - 1].replace(".blueprint", "")));
+                            partialPath = pack.getSubPath(blueprint.getFilePath().resolve(elements[elements.length - 1].replace(".blueprint", "")));
                         }
                         else
                         {
-                            partialPath = StructurePacks.getStructurePack(blueprint.getPackName()).getSubPath(blueprint.getFilePath().resolve(((IBlueprintDataProviderBE) be).getSchematicName()));
+                            partialPath = pack.getSubPath(blueprint.getFilePath().resolve(((IBlueprintDataProviderBE) be).getSchematicName()));
                         }
 
-                        if (!(world.getBlockEntity(centerPos) instanceof TileEntityColonyBuilding) && be instanceof TileEntityColonyBuilding)
+                        if (StructurePacks.findBlueprint(pack.getPath(), partialPath).isPresent())
```

Comment:
that's complicated, that's a lot of IO on thread

Reply:
What I would do here is:

We know the exact path it "should be at". So we can do: getBlueprintFuture with the ServerFutureProcessor andthen on success or failure set the values

Reply:
That's not straightforward either, this is in a conditional path and the following code can't happen until the blueprint path is set but the other branch of this condition shouldn't be delayed.

This would either need a new method on `ServerFutureProcessor` to kick something off to the background thread without having a specific blueprint in mind yet, or (better) a new method on `StructurePacks` that just checks for existence of the blueprint without actually loading it.

I've made an improvement on the assumption that the latter will eventually exist.

Reply:
Imo, this shouldn't be a problem at all to delay this, we return success always anyway.

Reply:
Ok, but "check if file exists" barely counts as an I/O operation.  It doesn't seem worth pushing it off to another thread.